src/Xrefcheck/System.hs

```diff
@@ -5,25 +5,43 @@
 
 module Xrefcheck.System
     ( readingSystem
-    , CanonicalizedGlobPattern (..)
+    , RelGlobPattern (..)
+    , bindGlobPattern
     ) where
 
 import Data.Aeson (FromJSON (..), withText)
 import GHC.IO.Unsafe (unsafePerformIO)
 import System.Directory (canonicalizePath)
 import qualified System.FilePath.Glob as Glob
+import System.FilePath.Posix ((</>))
```

Comment:
Why `Posix`? I mean, is it intentional and do you want it to still be `Posix` if someone tries to build it on Windows?

Reply:
Looks so, in glob patterns they (authors of that library) consider whatever filepath separator the system has, and `</>` also inserts `\\` in Windows.

Reply:
@Martoon-00 I am sorry, buy I still don't understand why `Posix`. And now that there was some effort to make this thing work on Windows, I'd like to get back to this question.

Just in case:
*    System.FilePath.Posix manipulates POSIX/Linux style FilePath values (with / as the path separator).
*    System.FilePath.Windows manipulates Windows style FilePath values (with either \ or / as the path separator, and deals with drives).
*    System.FilePath is an alias for the module appropriate to your platform.

So your `</>` is always `/`, even on Windows. You do: `canonicalizePath (root </> relPat)`. On Windows this `root </> relPat` can be something like `foo/bar` and `canonicalizePath` (I guess) won't treat `/` as a separator. So why is it done this way?

Reply:
Oh I got your point, that was really weird :facepalm:

Reply:
I inserted a commit prior to version bump. There are many places where import should be fixed.

Reply:
And that could be the reason why `qReadFileBS` didn't work and @balsoft had to replace it with `embedFile`, my bad.

Reply:
No, it's not indeed the reason. The problem with `qReadFileBS` lies deeper within the architecture of Template Haskell itself and it requires fixing it and not this.

(Obviously I tried replacing `Posix` with `Windows` here and it didn't help a bit)

Reply:
Ah okay, thanks for the note.